### PR TITLE
Add Blacklisted Account and App metrics

### PR DESCRIPTION
**Pull request by RossGoncharuk**
Added metrics for blacklisted accounts and app's.

### CODE REVIEW

File: src/main/java/org/prebid/server/exception/BlacklistedAccountOrApp.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.prebid.server.exception;
+
+public class BlacklistedAccountOrApp extends RuntimeException {
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/exception/BlacklistedAccountOrApp.java]: i'd like to assume separate exceptions.
Also, it is good practice to suffix with exception name `Exception`.

File: src/main/java/org/prebid/server/exception/InvalidRequestException.java [hunk 2]
```diff
@@ -20,7 +19,7 @@ public InvalidRequestException(String message, Throwable cause) {
     }
 
     public InvalidRequestException(List<String> messages) {
-        super(messages.stream().collect(Collectors.joining("\n")));
+        super(String.join("\n", messages));
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/exception/InvalidRequestException.java]: Nice :)

File: src/main/java/org/prebid/server/validation/RequestValidator.java [hunk 3]
```diff
@@ -360,11 +364,21 @@ private void validateSite(Site site) throws ValidationException {
...
-            try {
-                Json.mapper.treeToValue(app.getExt(), ExtApp.class);
-            } catch (JsonProcessingException e) {
-                throw new ValidationException("request.app.ext object is not valid: %s", e.getMessage());
+        if (app != null) {
+            final String appId = app.getId();
+            if (CollectionUtils.isNotEmpty(blacklistedApps) && StringUtils.isNotBlank(appId)
+                    && blacklistedApps.contains(appId)) {
+                // propagate BlacklistedAccOrApp instantly
+                throw new BlacklistedAccountOrApp(String.format(
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/validation/RequestValidator.java]: How `BlacklistedAccountOrApp` is catching?
Moreover, i think blacklist checking is not a part of request validation. Maybe it make sense to create separate component for such "security" actions. What do you think?

Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/validation/RequestValidator.java]: Well, it is NOT being caught all the way up to `handleResult` in `AuctionHandler`. As for "security" component - I don't think it is necessary at this moment, maybe if we get some more related aspects other than blacklisted apps and accounts. 

To me it is more of request-validation related moments, just that if we found either being in the blacklist - we should cancel the request immediately. So maybe it does make sense to check blacklists separately...

File: src/main/java/org/prebid/server/handler/openrtb2/AmpHandler.java [hunk 4]
```diff
@@ -287,7 +289,17 @@ private void handleResult(AsyncResult<AmpResponse> responseResult, AmpEvent.AmpE
             body = Json.encode(responseResult.result());
         } else {
             final Throwable exception = responseResult.cause();
-            if (exception instanceof InvalidRequestException) {
+            if (exception instanceof BlacklistedAppException || exception instanceof BlacklistedAccountException) {
+                metricRequestStatus = exception instanceof BlacklistedAccountException
+                        ? MetricName.blacklisted_account : MetricName.blacklisted_app;
+                final String errorMessage = exception.getMessage();
+                logger.info("Blacklisted: {0}", errorMessage);
```
Comment (schernysh) [inline src/main/java/org/prebid/server/handler/openrtb2/AmpHandler.java]: Should it really be logged with `info` level? It threatens that log file to be overwhelmed by such entries if some app or publisher starts shelling PBS with requests.

File: src/main/java/org/prebid/server/handler/openrtb2/AuctionHandler.java [hunk 5]
```diff
@@ -125,7 +127,17 @@ private void handleResult(AsyncResult<Tuple2<BidResponse, AuctionContext>> respo
             body = Json.encode(responseResult.result().getLeft());
         } else {
             final Throwable exception = responseResult.cause();
-            if (exception instanceof InvalidRequestException) {
+            if (exception instanceof BlacklistedAppException || exception instanceof BlacklistedAccountException) {
+                metricRequestStatus = exception instanceof BlacklistedAccountException
+                        ? MetricName.blacklisted_account : MetricName.blacklisted_app;
+                final String errorMessage = exception.getMessage();
+                logger.info("Blacklisted: {0}", errorMessage);
```
Comment (schernysh) [inline src/main/java/org/prebid/server/handler/openrtb2/AuctionHandler.java]: Same concern as above

File: src/main/java/org/prebid/server/handler/openrtb2/AmpHandler.java (hunk 4 above)
Comment (RossGoncharuk) [reply src/main/java/org/prebid/server/handler/openrtb2/AmpHandler.java]: Will then change it to `debug` as I've discussed with @rpanchyk

**Pull request merged**